### DISCUSSION TITLE
missing modules 4.8.1

### DISCUSSION THREAD

**Discussion by JohnRDOrazio** [WeblateOrg/weblate#6629] [Q&A] [answered]
I upgraded from Weblate 4.7 to Weblate 4.8.1 a couple days ago. The upgrade went fairly smoothly, however when running `weblate check --deploy` from within my python virtual environment, I get a couple warnings about missing modules:

> _ | _ | _
> -- | -- | --
> weblate.W025.stringsdict | Failure in loading handler for stringsdict file format: No module named 'translate.storage.stringsdict' | Documentazione
> weblate.W025.fluent | Failure in loading handler for fluent file format: No module named 'translate.storage.fluent'

I believe I have installed all the dependencies, I'm not sure how to go about installing these missing dependencies? Any pointers?

**Comment by nijel** [comment, marked as answer]
Upgrade translate-toolkit.

**Reply by JohnRDOrazio** [reply]
That did the trick. I issued a `pip install translate-toolkit[all] -U` within my virtual environment and this did in fact install a couple new components. After that my `weblate check --deploy` passes without those warnings (I had to restart my celery service, and just to be sure I also restarted my uwsgi service, otherwise I was getting this error: `The Celery process is outdated or misconfigured. Following items differ: formats`; after restarting services this error went away)

**Reply by JohnRDOrazio** [reply]
perhaps a note about updating **translate-toolkit** could be useful on the [weblate upgrade page](https://docs.weblate.org/en/latest/admin/upgrade.html)?

**Reply by nijel** [reply]
The tricky part is that we don't want to push people to install support for all formats translate-toolkit supports (some have quite heavy dependencies). Having a warning for a file format you are not going to use is okay. On the other side, we want to indicate that the user is missing some feature, that's why the warning is there.

**Reply by nijel** [reply]
I've made some clarifications in https://docs.weblate.org/en/latest/admin/install.html#optional-dependencies and https://docs.weblate.org/en/latest/admin/upgrade.html#generic-upgrade-instructions to better explain this:

* https://github.com/WeblateOrg/weblate/commit/5cca0eaa769581e5431b3c7c5deb55d7a5cac402
* https://github.com/WeblateOrg/weblate/commit/3fa65cf7e30847ac0470d44102a343d59faa31e5
* https://github.com/WeblateOrg/weblate/commit/c2e17bb3fb244a299722c449ded6f0b783debc97

**Reply by JohnRDOrazio** [reply]
am I understanding correctly that, to install new features while upgrading, one can issue `pip install -U Weblate[all]` instead of just simply `pip install -U Weblate` (as currently indicated in the **generic upgrade instructions**)? Or is it sufficient to update the `requirements-optional.txt` file before issuing `pip install -U Weblate`?

**Reply by nijel** [reply]
`Weblate[all]` will give you Weblate with all features.  With just `Weblate` you might miss some optional features.

Maybe it's better to recommend installing full-featured Weblate by default, I've done that in abb722c93dc37d38720280dced58a14627b43b04.
